packages/go_router/lib/src/match.dart

```diff
@@ -61,6 +61,29 @@ class RouteMatch {
     throw MatcherError('Unexpected route type: $route', restLoc);
   }
 
+  /// Completes the promise returned by [GoRouter.push], allowing comunication
+  /// between pages.
+  ///
+  /// If the promise has already been completed, this method does nothing.
+  ///
+  /// E.g.:
+  /// ```dart
+  /// final bool? result = await context.push<bool>('/page2');
+  /// if(result ?? false){
+  ///   // do something
+  /// }
+  /// ```
+  /// When the page is popped, the promise is completed with the value passed,
+  /// and the push method returns.
+  /// to [Navigator.pop].
+  /// ```dart
+  /// context.pop(true);
+  /// ```
+  void complete([dynamic value]) {}
```

Comment:
@chunhtai what do you think?

Reply:
why optional? I think it can be `Object? value`.

Reply:
I meant this method should be call `didComplete`.

void didComplete(Object? value) {}